src/wire/ipv6fragment.rs

```diff
@@ -0,0 +1,280 @@
+// |  Next Header  |   Reserved    |      Fragment Offset    |Res|M|
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |                         Identification                        |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//
+// See https://tools.ietf.org/html/rfc8200#section-4.4 for details.
+mod field {
+    use wire::field::*;
+
+    // 8-bit identifier of the header immediately following this header.
+    pub const NXT_HDR:          usize = 0;
+    // 8-bit reserved field.
+    pub const RESERVED:         usize = 1;
+    // 16-bit field containing the fragment offset, reserved and more fragments values.
+    pub const FR_OF_M:          Field = 2..4;
+    // 32-bit field identifying the fragmented packet
+    pub const IDENT:            Field = 4..8;
+}
+
+impl<T: AsRef<[u8]>> Header<T> {
+    /// Create a raw octet buffer with an IPv6 Fragment Header structure.
+    pub fn new(buffer: T) -> Header<T> {
+        Header { buffer }
+    }
+
+    /// Shorthand for a combination of [new] and [check_len].
+    ///
+    /// [new]: #method.new
+    /// [check_len]: #method.check_len
+    pub fn new_checked(buffer: T) -> Result<Header<T>> {
+        let header = Self::new(buffer);
+        header.check_len()?;
+        Ok(header)
+    }
+
+    /// Ensure that no accessor method will panic if called.
+    /// Returns `Err(Error::Truncated)` if the buffer is too short.
+    pub fn check_len(&self) -> Result<()> {
+        let data = self.buffer.as_ref();
+        let len = data.len();
+
+        if len < field::IDENT.end {
+            return Err(Error::Truncated);
+        }
+
+        Ok(())
+    }
+
+    /// Consume the header, returning the underlying buffer.
+    pub fn into_inner(self) -> T {
+        self.buffer
+    }
+
+    /// Return the next header field.
+    #[inline]
+    pub fn next_header(&self) -> Protocol {
+        let data = self.buffer.as_ref();
+        Protocol::from(data[field::NXT_HDR])
+    }
+
+    /// Return the fragment offset field.
+    #[inline]
+    pub fn frag_offset(&self) -> u16 {
+        let data = self.buffer.as_ref();
+        NetworkEndian::read_u16(&data[field::FR_OF_M]) >> 3
+    }
+
+    /// Return more fragment flag field.
+    #[inline]
+    pub fn more_frags(&self) -> bool {
+        let data = self.buffer.as_ref();
+        (data[3] & 0x1) == 1
+    }
+
+    /// Return the fragment identification value field.
+    #[inline]
+    pub fn ident(&self) -> u32 {
+        let data = self.buffer.as_ref();
+        NetworkEndian::read_u32(&data[field::IDENT])
+    }
+}
+
+impl<T: AsRef<[u8]> + AsMut<[u8]>> Header<T> {
+    /// Set the next header field.
+    #[inline]
+    pub fn set_next_header(&mut self, value: Protocol) {
+        let data = self.buffer.as_mut();
+        data[field::NXT_HDR] = value.into();
+    }
+
+    /// Set the reserved field.
+    #[inline]
+    pub fn set_reserved(&mut self, value: u8) {
```

Comment:
Let's have a dedicated function that clears all reserved fields to their default values instead.

Reply:
I made a single `set_reserved` function that sets both reserved fields to `0`.